plugins/woocommerce/src/Database/Migrations/CustomOrderTable/CLIRunner.php

```diff
@@ -0,0 +1,241 @@
+<?php
+
+namespace Automattic\WooCommerce\DataBase\Migrations\CustomOrderTable;
+
+use Automattic\WooCommerce\Internal\DataStores\Orders\CustomOrdersTableController;
+use Automattic\WooCommerce\Internal\DataStores\Orders\DataSynchronizer;
+use WP_CLI;
+
+/**
+ * Credits https://github.com/liquidweb/woocommerce-custom-orders-table/blob/develop/includes/class-woocommerce-custom-orders-table-cli.php.
+ *
+ * CLI tool for migrating order data to/from custom table.
+ *
+ * Class CLIRunner
+ */
+class CLIRunner {
+
+	/**
+	 * CustomOrdersTableController instance.
+	 *
+	 * @var CustomOrdersTableController
+	 */
+	private $controller;
+
+	/**
+	 * DataSynchronizer instance.
+	 *
+	 * @var DataSynchronizer;
+	 */
+	private $synchronizer;
+
+	/**
+	 * PostsToOrdersMigrationController instance.
+	 *
+	 * @var PostsToOrdersMigrationController
+	 */
+	private $post_to_cot_migrator;
+
+	/**
+	 * Init method, invoked by DI container.
+	 *
+	 * @param CustomOrdersTableController      $controller Instance.
+	 * @param DataSynchronizer                 $synchronizer Instance.
+	 * @param PostsToOrdersMigrationController $posts_to_orders_migration_controller Instance.
+	 *
+	 * @internal
+	 */
+	final public function init( CustomOrdersTableController $controller, DataSynchronizer $synchronizer, PostsToOrdersMigrationController $posts_to_orders_migration_controller ) {
+		$this->controller           = $controller;
+		$this->synchronizer         = $synchronizer;
+		$this->post_to_cot_migrator = $posts_to_orders_migration_controller;
+	}
+
+	/**
+	 * Check if the COT feature is enabled.
+	 *
+	 * @param bool $log Optionally log a error message.
+	 *
+	 * @return bool Whether the COT feature is enabled.
+	 */
+	private function is_enabled( $log = true ) {
+		if ( ! $this->controller->is_feature_visible() ) {
+			if ( $log ) {
+				WP_CLI::log( __( 'Custom order table usage is not enabled.', 'woocommerce' ) );
+			}
+		}
+
+		return $this->controller->is_feature_visible();
+	}
+
+	/**
+	 * Helper method to log warning that feature is not yet production ready.
+	 */
+	private function log_product_warning() {
```

Comment:
Addressed in https://github.com/woocommerce/woocommerce/pull/32938/commits/124fdd37f35389dd7ac8ba92255bb159dbc26b9b. Although didn't added a check for `wp_get_environment_type` as it felt like it would be one more hurdle in getting test results, wdyt?

Reply:
Yup, that sounds reasonable. 👍